dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/map/widgets/D3Map.java

```diff
@@ -0,0 +1,260 @@
+package org.dashbuilder.renderer.c3.client.charts.map.widgets;
+
+import java.util.IntSummaryStatistics;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import org.dashbuilder.renderer.c3.client.charts.map.D3MapConf;
+import org.dashbuilder.renderer.c3.client.charts.map.GeoData;
+import org.dashbuilder.renderer.c3.client.jsbinding.d3.D3PathGenerator;
+import org.dashbuilder.renderer.c3.client.jsbinding.d3.D3Projection;
+import org.jboss.errai.common.client.api.elemental2.IsElement;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+import org.uberfire.client.views.pfly.widgets.D3;
+import org.uberfire.client.views.pfly.widgets.D3.Selection;
+
+import com.google.gwt.core.client.JavaScriptObject;
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONValue;
+
+import elemental2.dom.DomGlobal;
+import elemental2.dom.Element;
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLElement;
+
+@Templated
+@Dependent
+public class D3Map implements IsElement {
+    
+    private static final int RADIUS_PROPORTION = 20;
+    private static final int MIN_RADIUS= 2;
+    
+    private int width;
+    private int height;
+    private D3MapConf conf;
+    private D3.Scale colorScale;
+    private GeoData geoData;
+    private Map<String, Double> data;
+    
+    @Inject
+    private MapTooltip mapTooltip;
+    
+    @Inject
+    private HTMLDivElement mapContainer;
+
+    D3 d3 = D3.Builder.get();
+    String[] COLORS_SCHEME = d3.getSchemeGreens()[7];
```

Comment:
Could you please extract this magical nubmer into constant? This way it's kind of hard to tell what it means..